libraries/abstractions/transport/secure_sockets/transport_secure_sockets.c

```diff
@@ -0,0 +1,507 @@
+    {
+        TransportSocketStatus = SOCKETS_SetSockOpt( tcpSocket,
+                                                    0,
+                                                    SOCKETS_SO_ALPN_PROTOCOLS,
+                                                    ppcALPNProtos,
+                                                    sizeof( ppcALPNProtos ) / sizeof( ppcALPNProtos[ 0 ] ) );
+
+        if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+        {
+            LogError( ( "Failed to set ALPN option socket." ) );
+        }
+    }
+
+    /* Set SNI option. */
+    if( ( TransportSocketStatus == SOCKETS_ERROR_NONE ) && ( pSocketsConfig->disableSni == false ) )
+    {
+        TransportSocketStatus = SOCKETS_SetSockOpt( tcpSocket,
+                                                    0,
+                                                    SOCKETS_SO_SERVER_NAME_INDICATION,
+                                                    pHostName,
+                                                    hostnameLength + 1 );
+
+        if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+        {
+            LogError( ( "Failed to set SNI option for socket. TransportSocketStatus %d, hostnameLength  + 1 %d ",
+                        TransportSocketStatus, hostnameLength + 1 ) );
+        }
+    }
+
+    /* Set custom server certificate. */
+    if( ( TransportSocketStatus == SOCKETS_ERROR_NONE ) && ( pSocketsConfig->pRootCa != NULL ) )
+    {
+        TransportSocketStatus = SOCKETS_SetSockOpt( tcpSocket,
+                                                    0,
+                                                    SOCKETS_SO_TRUSTED_SERVER_CERTIFICATE,
+                                                    pSocketsConfig->pRootCa,
+                                                    pSocketsConfig->rootCaSize );
+
+        if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+        {
+            LogError( ( "Failed to set server certificate option for socket." ) );
+        }
+    }
+
+    return TransportSocketStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t transportTimeoutSetup( Socket_t tcpSocket,
+                                      uint32_t sendTimeoutMs,
+                                      uint32_t recvTimeoutMs )
+{
+    TickType_t receiveTimeout = 0, sendTimeout = 0;
+    int32_t TransportSocketStatus = ( int32_t ) SOCKETS_ERROR_NONE;
+
+    if( tcpSocket == NULL )
+    {
+        TransportSocketStatus = SOCKETS_EINVAL;
+    }
+
+    if( TransportSocketStatus == SOCKETS_ERROR_NONE )
+    {
+        /* Secure Sockets uses TickType_t therefore replace timeout vale with portMAX_DELAY if it exceeds. */
+        if( pdMS_TO_TICKS( recvTimeoutMs ) > portMAX_DELAY )
+        {
+            receiveTimeout = portMAX_DELAY;
+        }
+        else
+        {
+            receiveTimeout = pdMS_TO_TICKS( recvTimeoutMs );
+        }
```

Comment:
How about this for a single call to `pdMS_TO_TICKS`?
```suggestion
        receiveTimeout =  pdMS_TO_TICKS( recvTimeoutMs );
        if(  pdMS_TO_TICKS( recvTimeoutMs ) > portMAX_DELAY )
        {
             receiveTimeout = portMAX_DELAY;
        }
```